module-workers.tf

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  worker_count_expected = coalesce(one(module.workers[*].worker_count_expected), 0)
+
+  # Distinct list of compartments for enabled worker pools
+  worker_compartments = distinct(compact([
+    for k, v in var.worker_pools : lookup(v, "compartment_id", local.compartment_id)
+    if tobool(lookup(v, "enabled", var.worker_pool_enabled))
+  ]))
+
+  # Worker pools with cluster autoscaler management enabled
+  autoscaler_compartments = distinct(compact([
+    for k, v in var.worker_pools : lookup(v, "compartment_id", local.compartment_id)
+    if tobool(lookup(v, "enabled", var.worker_pool_enabled)) && tobool(lookup(v, "allow_autoscaler", false))
+  ]))
+
+  # Distinct list of compartments for cluster autoscaler-enabled worker pools
+  autoscaling_groups = var.create_cluster ? {
+    for k, v in one(module.workers[*].worker_pools) : k => v if tobool(lookup(v, "autoscale", false))
+  } : {}
+}
+
+# Default workers sub-module implementation for OKE cluster
+module "workers" {
+  count  = var.create_cluster ? 1 : 0
+  source = "./modules/workers"
+
+  # Common
+  compartment_id      = local.worker_compartment_id
+  tenancy_id          = local.tenancy_id
+  state_id            = random_id.state_id.id
+  ad_numbers          = local.ad_numbers
+  ad_numbers_to_names = local.ad_numbers_to_names
+
+  # Cluster
+  apiserver_private_host = try(split(":", one(module.cluster[*].endpoints.private_endpoint))[0], "")
+  cluster_ca_cert        = local.cluster_ca_cert
+  cluster_dns            = var.cluster_dns
+  cluster_id             = coalesce(var.cluster_id, one(module.cluster[*].cluster_id))
+  kubernetes_version     = var.kubernetes_version
+
+  # Worker pools
+  worker_pool_enabled = var.worker_pool_enabled
+  worker_pool_mode    = var.worker_pool_mode
+  worker_pool_size    = var.worker_pool_size
+  worker_pools        = var.worker_pools
+
+  # Workers
+  assign_dns            = var.assign_dns
+  assign_public_ip      = var.worker_type == "public"
+  block_volume_type     = var.worker_block_volume_type
+  cloudinit             = var.worker_cloudinit
```

Comment:
How are you allowing for multiple cloud-inits?

Reply:
Workers/operator allow specification via `list(map(string))` of MIME multi-parts - the [variable descriptions](https://github.com/oracle-terraform-modules/terraform-oci-oke/pull/640/files#diff-cc4febfbdbdf88d60319cc80a7dba73411d1effd41defdcc8d922a5410915ff3R150) and [precondition validations](https://github.com/oracle-terraform-modules/terraform-oci-oke/pull/640/files#diff-ebb04e6baff3b5ecfb921f4f51dbaeadc6bcebb02c26415c27fef07ffba550d3R100-R142) reference cloud-init docs for the schema and configuration, and additional examples will be included with our own documentation.